sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java

```diff
@@ -258,4 +282,16 @@ private InputFile inputFile(String name) {
     return inputFile;
   }
 
+  private void verifyUsages(String componentKey, int line, int offset, TextPointer... tps) {
+    Collection<TextRange> textRanges = context.referencesForSymbolAt(componentKey, line, offset);
+    if (tps.length == 0) {
+      assertThat(textRanges).isEmpty();
+    } else {
+      assertThat(textRanges.stream().map(TextRange::start).collect(Collectors.toList())).isNotEmpty().containsOnly(tps);
```

Comment:
I don't think you need to collect to assert on a stream.  And moreover, the containsOnly assertion will check it is not Empty, so no need for the isNotEmpty call.

Reply:
Please note that you only tested START of references and you most probably have a bug of end of references.